pkgs/pkgs-lib/formats.nix

```diff
@@ -76,10 +76,11 @@ rec {
 
   };
 
-  yaml = {}: {
+  yaml = lib.warn "`pkgs.formats.yaml` is deprecated in favor of explicitly versioned YAML formatters instead. Use `pkgs.formats.yaml_1_1` for YAML 1.1." yaml_1_1;
```

Comment:
just an idea (maybe not a clever one):

have a default-parameter `format ? "1.1` in place and reuse the the existing handle? Then we do not need to rename all existing `formats.yaml`

Reply:
If we had the module system, and I could specify an enum type, then maybe. But we don't so I think this makes the UX worse.

Reply:
sure. I agree